trans/src/Pact/Trans/Types.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveFoldable #-}
+{-# LANGUAGE DeriveTraversable #-}
+
+-- |
+-- Module      :  Pact.Trans.Types
+-- Copyright   :  (C) 2023 John Wiegley
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  John Wiegley <john@kadena.io>
+--
+-- Operators and math built-ins based on the MPFR library.
+--
+
+module Pact.Trans.Types
+  ( TransResult(..)
+  , doubleToTransResult
+  ) where
+
+data TransResult a
+  = TransNumber !a
+  | TransNaN !a
+  | TransInf !a
+  | TransNegInf !a
+  deriving (Show, Eq, Ord, Functor, Foldable, Traversable)
+
```

Comment:
Should we provide `isNaN`, `isInf`, ....  for `TransResult`?

Reply:
Great idea

Reply:
Done